### ISSUE TITLE
getBytes is deprecated, but getBlob is not supported by Oracle

### ISSUE THREAD

**Issue by gsheasby** [palantir/atlasdb#2646]
We have had a few cases (fixed in #2221 and #2645)  where Oracle tests have failed, because `AgnosticLightResultRow.getBlob` is not supported by the Oracle implementation. The fix for the tests is to use `.getBytes`, which is `@Deprecated`.

The exception we get is: `Caused by: java.sql.SQLException: Invalid column type: getBLOB not implemented for class oracle.jdbc.driver.T4CRawAccessor`

To properly fix this, we need to do one of the following:
a) un-deprecate `getBytes`
b) Fix `getBlob` so that it does not lead us down the path towards `getBLOB`
c) Upgrade our internal Oracle driver to a version that supports `getBLOB`.

**Comment by gsheasby** [comment]
Problematic AtlasDB code:
ResultSets.getBlob(ResultSet, int):
```
    public static Blob getBlob(ResultSet rs, int col) throws PalantirSqlException {
        try {
            return rs.getBlob(col);
        } catch (SQLException e) {
            throw BasicSQL.handleInterruptions(0, e);
        }
    }
```

AgnosticLightResultRowImpl.getBlob(int, int):
```
    public byte [] getBlob(int col, int blobLength) throws PalantirSqlException {
        if(dbType == DBType.ORACLE) {
            Blob contentBlob = ResultSets.getBlob(results, col);
```

**Comment by hsaraogi** [comment]
@dxiao for thoughts?

**Comment by dxiao** [comment]
surprised that it's not supported by oracle, we use getBlob internally in a lot of places - is the problem that getBlob is for BLOB types and getBytes is for RAW types? In which case, this is probably more an issue of documentation than deprecation and changes. Probably my mistake for originally confusing the two in the changeset I had before.

**Comment by schlosna** [comment]
yeah, the problem is trying to use `getBlob` on a `RAW` binary column type.
